lib/instrumentation/amqplib/amqplib.js

```diff
@@ -266,18 +266,20 @@ function wrapModel(shim, Model, promiseMode) {
       destinationName: shim.FIRST,
       callback: setCallback(shim, promiseMode),
       promise: promiseMode,
-      messageHandler: function handleConsumedMessage(shim, fn, name, message) {
+      after: function handleConsumedMessage({ shim, result, args, segment }) {
+        if (!shim.agent.config.message_tracer.segment_parameters.enabled) {
+          shim.logger.trace('Not capturing segment parameters')
+          return
+        }
+
         // the message is the param when using the promised based model
-        message = promiseMode ? message : message[1]
+        const message = promiseMode ? result : args?.[1]
         if (!message) {
           shim.logger.trace('No results from consume.')
           return null
         }
         const parameters = getParametersFromMessage(message)
-
-        const headers = message?.properties?.headers
-
-        return { parameters, headers }
```

Comment:
headers were never getting used here. I _could_ mimic what's being done in [here](https://github.com/newrelic/node-newrelic/blob/main/lib/shim/message-shim/subscribe-consume.js#L185) but we never had tests that asserting DT headers were getting propagated. I would _think_ we should. so maybe i'll cut another story to fix this

Reply:
ok i dug into this and I don't think it's apples to apples in `recordSubscribecConsume` it creates a transaction for the consumption. Whereas `.get` is just pulling one message.  I can't just add the headers to DT payload as there's only 1 transaction getting created elsewhere and not within `.get`. So I'll leave this as is

Reply:
Should that be noted in an inline comment?

Reply:
I'm not sure because there's no code to say we don't need to propagate headers